File: src/main/scala/ContainerWithMostWater.scala
object ContainerWithMostWater extends App {
  def maxArea(height: Array[Int]): Int = {
      if(height == null || height.length < 2) 0
      else {
         var p = 0
         var q = height.length - 1
         var maxCapacity = Int.MinValue
         while(p < q) {
            val h = height(q) min height(p)
            maxCapacity = (h * (q - p)) max maxCapacity
            if(height(p) < height(q)) p += 1
            else q -=1
         }
         maxCapacity
      }
  }

  val h = Array(1,8,6,2,5,4,8,3,7)
  println(maxArea(h))
}

File: src/main/scala/algorithms/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height) -> int:
        if len(height) < 0:
            return 0
        if len(height) == 2:
            return min(height) * 1
        max_cap = 0
        l = 0
        r = len(height) - 1
        while l < r:
            max_cap = max(max_cap, (r - l) * min(height[r], height[l]) )
            if height[l] < height[r]:
                l += 1
            elif height[l] > height[r]:
                r -= 1
            else:
                l += 1
                r -= 1
        return max_cap
